velox/functions/sparksql/aggregates/ModeAggregate.cpp

```diff
@@ -0,0 +1,585 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/aggregates/ModeAggregate.h"
+
+#include "folly/container/F14Map.h"
+
+#include "velox/exec/AddressableNonNullValueList.h"
+#include "velox/exec/SimpleAggregateAdapter.h"
+#include "velox/exec/Strings.h"
+
+using namespace facebook::velox::aggregate::prestosql;
+using namespace facebook::velox::exec;
+
+namespace facebook::velox::functions::aggregate::sparksql {
+
+namespace {
+
+struct UnknownValueHash {
+  std::size_t operator()(const UnknownValue& val) const noexcept {
+    return folly::hasher<UnknownValue>{}(val);
```

Comment:
We could use std::hash(null) for Unknown type, it is all null in Spark mapping to NullType

Reply:
I think this is just a placeholder, it would not be executed, as nulls would be filtered.